### DISCUSSION TITLE
Could the "currency" field be changed into a dropdown? Is that technically feasible?

### DISCUSSION THREAD

**Discussion by woutersamaey** [btcpayserver/btcpayserver#2602] [Ideas / Feature Requests]
Could the "currency" field be changed into a dropdown? Is that technically feasible?

![Image Pasted at 2021-6-21 15-56](https://user-images.githubusercontent.com/71019/122774829-d765a180-d2a9-11eb-88da-b863bcdc8ba7.png)

If so, we could use this. Looks way better and more compact. 

![Image Pasted at 2021-6-21 15-58](https://user-images.githubusercontent.com/71019/122774840-d92f6500-d2a9-11eb-8912-337126c640ca.png)

Also, no need to study all the currency codes as we could show a human friendly label and have the value internally

**Comment by woutersamaey** [comment]
@dennisreimann and @dstrukt what do you think?

**Comment by pavlenex** [comment]
I believe @NicolasDorier had a good argument against the currency dropdown once I first suggested it a while ago. I am curious if that argument is still valid?

@woutersamaey Could you elaborate where exactly you would like to see the dropdown if technically feasible? 

I think we should find a way to display them in a dropdown and list all currencies with an auto-populate option, as iI noticed it's confusing for people to know what they have to type in.

**Reply by woutersamaey** [reply]
Well, the create invoice form now uses 2 fields for amount + currency.
If we can have the currency as a dropdown, these 2 fields could be presented as 1 like in the screenshot. Bootstrap supports this kind of presentation and it looks nice.
Besides EUR, USD and BTC, I have trouble thinking of other currency codes. A list would definately help people see what are valid currencies and what are not.
I understand there may be 100s of currencies in the world, but most shops would only use 2 or 3 depending on their location.
We can make this more slick (visually and usability wise).

**Reply by dennisreimann** [reply]
I like this idea and we could build a custom component that progressively enhances what we currently have to something like this:

![amount](https://user-images.githubusercontent.com/886/122965172-3568b680-d388-11eb-8626-f95736a28898.png)

I think this would offer guidance as well as a better visual representation.

**Reply by woutersamaey** [reply]
This is exactly what i had in mind.

**Reply by dstrukt** [reply]
Also agree this is something we should support from a UX standpoint if it's technically feasible, much easier to grok than manually inputting your currency.

In our mocks as well.

ACK.

**Reply by woutersamaey** [reply]
@dennisreimann do you think we should add the currency symbol to the dropdown?
EUR and USD look good, but SATS looks a bit strange. Maybe change to "Bitcoin (sats)" ? There is no need to keep using 3-4 letter codes if we have a dropdown.
And, what about Litecoin sats and other crypto? Do they all get 3-4 letter codes? 🤔
What about "Bits" and "Millibitcoin" etc... Any thoughts on that?

**Reply by pavlenex** [reply]
Good suggestion @dennisreimann 

I am against adding symbols as it may lead people to think they can write $ in the field to create an invoice for a particular currency.

@woutersamaey We don't support bits and milibits only BTC and SATS afair.

**Comment by woutersamaey** [comment]
If proved feasable, I'd also like to see a currency list response in the Greenfield API like we currently have an endpoint for all languages, we should also have a API endpoint for all currencies.

The "All Currencies" endpoint could also contain basic exchange rates. Something Greenfield does not have ATM.

@Kukks do you think this is possible?

**Reply by dstrukt** [reply]
Pinging @Kukks again.

Also very curious to know if this possible.

**Comment by Kukks** [comment]
This is not entirely possible. We would need to have the dropdown while also having a freeform text field option. This is because currencies can be dynamic beyond our own hardcoded list as they come from the rates providers. (e.g. Kraken lists Bitcoin as XBT, so if you use them as the rates provider for your store, XBT is available to be used.)

**Reply by woutersamaey** [reply]
Looking at https://user-images.githubusercontent.com/886/122965172-3568b680-d388-11eb-8626-f95736a28898.png you will see we do have a dropdown + free text field.

**Reply by bolatovumar** [reply]
I guess we could have a dropdown with currencies which we know will be supported for sure like BTC, USD, EUR (?), etc... AND still allow the user to input something custom. This way for most user cases we are covered with just the dropdown and we save some typing and we don't lose any functionality.

**Reply by woutersamaey** [reply]
@bolatovumar exactly!

**Reply by dennisreimann** [reply]
@bolatovumar yes, I proposed this [here](https://github.com/btcpayserver/btcpayserver/discussions/2602#discussioncomment-906111) too.
